### PR TITLE
Show facehugger limit when examining eggs/morphers

### PR DESCRIPTION
**Pull request by zzzmike**
# About the pull request

When you examine a hive core, it tells you the lesser drone limit, and the current amount of lesser drones. This adds a similar behavior for eggs and morphers with regards to playable facehuggers.

# Explain why it's good for the game

Helpful for ghosts to decide whether to play hugger, drone, neither, or just for information.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
qol: examining eggs/morphers tells you current number of huggers and the limit
/:cl:

### CODE REVIEW

File: code/modules/cm_aliens/structures/egg.dm [hunk 1]
```diff
@@ -55,6 +55,10 @@
 	. = ..()
 	if(isxeno(user) && status == EGG_GROWN)
 		. += "Ctrl + Click egg to retrieve child into your empty hand if you can carry it."
+	if(isobserver(user) && status == EGG_GROWN)
+		var/datum/hive_status/hive = GLOB.hive_datum[XENO_HIVE_NORMAL]
```
Comment (harryob) [inline code/modules/cm_aliens/structures/egg.dm:59]:
```suggestion
		var/datum/hive_status/hive = GLOB.hive_datum[hivenumber]
```

File: code/modules/cm_aliens/structures/special/egg_morpher.dm [hunk 2]
```diff
@@ -44,6 +44,10 @@
 	. = ..()
 	if(isxeno(user) || isobserver(user))
 		. += "It has [stored_huggers] facehuggers within, with [huggers_to_grow] more to grow (reserved: [huggers_reserved])."
+	if(isobserver(user))
+		var/datum/hive_status/hive = GLOB.hive_datum[XENO_HIVE_NORMAL]
+		var/current_hugger_count = hive.get_current_playable_facehugger_count();
+		. += "Current facehugger limit: <b>[current_hugger_count]/[hive.playable_hugger_limit]</b>"
```
Comment (harryob) [inline code/modules/cm_aliens/structures/special/egg_morpher.dm:50]:
```suggestion
		var/current_hugger_count = linked_hive.get_current_playable_facehugger_count();
		. += "Current facehugger limit: <b>[current_hugger_count]/[hive.playable_hugger_limit]</b>"
```

File: code/modules/mob/living/carbon/xenomorph/hive_status.dm [hunk 3]
```diff
@@ -874,6 +874,14 @@
 
 	return TRUE
 
+/datum/hive_status/proc/get_current_playable_facehugger_count()
+	var/count = 0
+	for(var/mob/mob as anything in totalXenos)
+		if(isfacehugger(mob))
+			count++
+	return count
+endproc
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/hive_status.dm:883]: huh?

```suggestion
```

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/hive_status.dm [hunk 4]
```diff
@@ -874,6 +874,13 @@
 
 	return TRUE
 
+/datum/hive_status/proc/get_current_playable_facehugger_count()
+	var/count = 0
+	for(var/mob/mob as anything in totalXenos)
+		if(isfacehugger(mob))
```
Comment (vero5123) [inline code/modules/mob/living/carbon/xenomorph/hive_status.dm:880]:
```suggestion
	for(var/mob/xeno_mob as anything in totalXenos)
		if(isfacehugger(xeno_mob))
```

File: code/modules/cm_aliens/structures/egg.dm [hunk 5]
```diff
@@ -55,6 +55,10 @@
 	. = ..()
 	if(isxeno(user) && status == EGG_GROWN)
 		. += "Ctrl + Click egg to retrieve child into your empty hand if you can carry it."
+	if(isobserver(user) && status == EGG_GROWN)
+		var/datum/hive_status/hive = GLOB.hive_datum[hivenumber]
+		var/current_hugger_count = hive.get_current_playable_facehugger_count();
+		. += "Current facehugger limit: <b>[current_hugger_count]/[hive.playable_hugger_limit]</b>"
```
Comment (Drulikar) [inline code/modules/cm_aliens/structures/egg.dm:61]:
```suggestion
		. += "There are currently [SPAN_NOTICE("[current_hugger_count]")] facehuggers in the hive. The hive can support [SPAN_NOTICE("[hive.playable_hugger_limit]")] facehuggers."
```

File: code/modules/cm_aliens/structures/special/egg_morpher.dm [hunk 6]
```diff
@@ -44,6 +44,9 @@
 	. = ..()
 	if(isxeno(user) || isobserver(user))
 		. += "It has [stored_huggers] facehuggers within, with [huggers_to_grow] more to grow (reserved: [huggers_reserved])."
+	if(isobserver(user))
+		var/current_hugger_count = linked_hive.get_current_playable_facehugger_count();
+		. += "Current facehugger limit: <b>[current_hugger_count]/[linked_hive.playable_hugger_limit]</b>"
```
Comment (Drulikar) [inline code/modules/cm_aliens/structures/special/egg_morpher.dm:49]:
```suggestion
		. += "There are currently [SPAN_NOTICE("[current_hugger_count]")] facehuggers in the hive. The hive can support [SPAN_NOTICE("[linked_hive.playable_hugger_limit]")] facehuggers."
```

**Review by Drulikar — CHANGES_REQUESTED** [review]
To use the same formatting as lesser examines.

image

**Review by Drulikar — APPROVED** [review]

**Pull request merged**